src/analysis/normed_space/extend.lean

```diff
@@ -0,0 +1,161 @@
+
+/-- Extend `fr : F →ₗ[ℝ] ℝ` to `F →ₗ[ℂ] ℂ` in a way that will also be continuous and have its norm
+bounded by `∥fr∥` if `fr` is continuous. -/
+noncomputable def linear_map.extend_to_ℂ (fr : F →ₗ[ℝ] ℝ) : F →ₗ[ℂ] ℂ :=
+begin
+  let fc : F → ℂ := λ x, ⟨fr x, -fr (I • x)⟩,
+  have add : ∀ x y : F, fc (x + y) = fc x + fc y,
+  {
+    intros,
+    ext; dsimp,
+    { rw fr.map_add },
+    rw [smul_add, fr.map_add, neg_add],
+  },
+
+  have smul_ℝ : ∀ (c : ℝ) (x : F), fc (c • x) = c * fc x,
+  {
+    intros,
+    ext; dsimp,
+    { rw [fr.map_smul, smul_eq_mul, zero_mul, sub_zero] },
+
+    rw [zero_mul, add_zero],
+    calc -fr (I • c • x)
+        = -fr (I • (c : ℂ) • x) : rfl
+    ... = -fr ((c : ℂ) • I • x) : by rw smul_comm
+    ... = -fr (c • I • x) : rfl
+    ... = -(c * fr (I • x)) : by rw [fr.map_smul, smul_eq_mul]
+    ... = c * -fr (I • x) : by rw neg_mul_eq_mul_neg,
+  },
+
+  have smul_I : ∀ x : F, fc (I • x) = I * fc x,
+  {
+    intros,
+    have h : -fr (I • I • x) = fr x,
+    {
+      calc -fr (I • I • x)
+          = -fr (((-1 : ℝ) : ℂ) • x) : by rw [←mul_smul, I_mul_I, of_real_neg, of_real_one]
+      ... = -fr ((-1 : ℝ) • x) : rfl
+      ... = -((-1 : ℝ) * fr x : ℝ) : by rw [fr.map_smul, smul_eq_mul]
+      ... = fr x : by ring,
+    },
+
+    calc fc (I • x)
+        = ⟨fr (I • x), -fr (I • I • x)⟩ : rfl
+    ... = ⟨fr (I • x), fr x⟩ : by rw h
+    ... = I * ⟨fr x, -fr (I • x)⟩ : by rw I_mul
+    ... = I * fc x : rfl,
+  },
+
+  have smul_ℂ : ∀ (c : ℂ) (x : F), fc (c • x) = c • fc x,
+  {
+    intros,
+    let a : ℂ := c.re,
+    let b : ℂ := c.im,
+    have h : fc ((b * I) • x) = b * I * fc x,
+    {
+      calc fc ((b * I) • x)
+          = fc (b • I • x) : by rw mul_smul
+      ... = fc (c.im • I • x) : rfl
+      ... = b * fc (I • x) : by rw smul_ℝ
+      ... = b * (I * fc x) : by rw smul_I
+      ... = b * I * fc x : by rw mul_assoc
+    },
+    calc fc (c • x)
+        = fc ((a + b * I) • x) : by rw re_add_im
+    ... = fc (a • x + (b * I) • x) : by rw add_smul
+    ... = fc (a • x) + fc ((b * I) • x) : by rw add
+    ... = fc (c.re • x) + fc ((b * I) • x) : rfl
+    ... = a * fc x + fc ((b * I) • x) : by rw smul_ℝ
+    ... = a * fc x + b * I * fc x : by rw h
+    ... = (a + b * I) * fc x : by rw add_mul
+    ... = c * fc x : by rw re_add_im c,
+  },
+
+  exact { to_fun := fc, map_add' := add, map_smul' := smul_ℂ }
+end
+
+/-- The norm of the extension is bounded by ∥fr∥. -/
+lemma norm_bound (fr : F →L[ℝ] ℝ) :
+  ∀ x : F, ∥fr.to_linear_map.extend_to_ℂ x∥ ≤ ∥fr∥ * ∥x∥ :=
+begin
+  intros,
+  let lm := fr.to_linear_map.extend_to_ℂ,
+
+  -- We aim to find a `t : ℂ` such that
+  -- * `lm (t • x) = fr (t • x)` (so `lm (t • x) = t * lm x ∈ ℝ`)
+  -- * `∥lm x∥ = ∥lm (t • x)∥` (so `t.abs` must be 1)
+  -- If `lm x ≠ 0`, `(lm x)⁻¹` satisfies the first requirement, and after normalizing, it
+  -- satisfies the second.
+  -- (If `lm x = 0`, the goal is trivial.)
+  classical,
+  by_cases lm x = 0,
```

Comment:
```suggestion
  by_cases h : lm x = 0,
```
You shouldn't use autogenerated names in proofs: if you use it, specify the name when it is generated.